code/game/objects/structures/barricade.dm

```diff
@@ -398,6 +398,9 @@
 			balloon_alert(user, "You need more wood")
 			return
 
+		if(LAZYACCESS(user.do_actions, src))
+			return
```

Comment:
can just check do_actions, otherwise you can still repair multiple cades at once with plasteel/metal.

Reply:
So I preserve current behaviour? This is how it currently works.